bmc/engine/verify_settling_time.h

```diff
@@ -0,0 +1,1436 @@
+                        double C[MAX_SIZE][MAX_SIZE], double D[MAX_SIZE][MAX_SIZE],
+                        double X0[MAX_SIZE][MAX_SIZE], double u, double tsr,
+                        double p, double ts, int dim)
+{
+  double yss, mp, lambMax, cbar, output, inf, sup;
+  PKVL out;
+  int kbar, kp, i;
+  xk.lastState = 0;
+  yss = y_ss(A, B, C, D, u, dim);
+  out = peak_output(A, B, C, D, X0, yss, u, dim);
+  mp = out.mp;
+  kp = out.kp;
+  lambMax = max_mag_eigenvalue(A, dim);
+  cbar = c_bar(mp, yss, lambMax, kp);
+  kbar = k_bar(lambMax, p, cbar, yss, dim);
+  #ifdef DEBUG
+  printf("Mp=%f\n", mp);
+  printf("yss=%f\n", yss);
+  printf("lambMax=%f\n", lambMax);
+  printf("kp=%d\n", kp);
+  printf("cbar=%f\n", cbar);
+  #endif
+  if(kbar * ts < tsr)
+  {
+    #ifdef DEBUG
+    printf("kbar=%d\n", kbar);
+    #endif
+    return 1;
+  }
+  i = (int)sp_ceil(tsr / ts)-1;
+  output = y_k3(A, B, C, D, u, i, X0, dim);
+  while(i <= kbar)
+  {
+    if(yss > 0)
+    {
+      inf = (yss - (yss * (p/100)));
+      sup = (yss * (p/100) + yss);
+    }
+    else
+    {
+      sup = (yss - (yss * (p/100)));
+      inf = (yss * (p/100) + yss);
+    }
+    if(!(output <= sup && (output >= inf)))
+    {
+      #ifdef DEBUG
+      printf("kbar=%d\n", kbar);
+      #endif
+      return 0;
+    }
+    i++;
+    output = y_k2(A, B, C, D, u, i, dim);
+  }
+  return 1;
+}
+
+int verify_settling_time(void)
+{
+  double ts, tsr, p, u;
+  int i, j, kbar, k_ss;
+  int nStates = 5;
+  double A[MAX_SIZE][MAX_SIZE], B[MAX_SIZE][MAX_SIZE], C[MAX_SIZE][MAX_SIZE];
+  double D[MAX_SIZE][MAX_SIZE], X0[MAX_SIZE][MAX_SIZE];
+  fxp_t min_fxp = fxp_double_to_fxp(impl.min);
+  fxp_t max_fxp = fxp_double_to_fxp(impl.max);
+  tsr = _controller.tsr;
+  ts = _controller.ts;
+  p = _controller.p;
+  u = (double)_controller.inputs[0][0];
+  double mynondet = nondet_double();
+  __DSVERIFIER_assume(
+      (mynondet >= min_fxp) &&
+      (mynondet <= max_fxp));
+  // Parsing matrices
+  // Matrix A
+  for(i = 0;i < nStates;i++)
+  {
+	for(j = 0;j < nStates;j++)
+	{
+      A[i][j] = _controller.A[i][j];
+	}
+  }
+  // Matrix B
+  for(i = 0;i < nStates;i++)
+  {
+    for(j = 0;j < 1;j++)
+  	{
+      B[i][j] = _controller.B[i][j];
+  	}
+  }
+  // Matrix C
+  for(i = 0;i < 1;i++)
+  {
+  	for(j = 0;j < nStates;j++)
+  	{
+      C[i][j] = _controller.C[i][j];
+  	}
+  }
+  // Matrix D
+  for(i = 0;i < 1;i++)
+  {
+  	for(j = 0;j < 1;j++)
+  	{
+      D[i][j] = _controller.D[i][j];
+  	}
+  }
+  // Matrix X0
+  for(i = 0;i < 1;i++)
+  {
+   	for(j = 0;j < nStates;j++)
+   	{
+      X0[i][j] = _controller.x0[i][j];
+  	}
+  }
+//  printf("Here -> A[0][1]=\n", A.me[0][1]);
+//  __DSVERIFIER_assert_msg(check_settling_time(A, B, C, D, X0, u, tsr, p, ts), "aqui");
+  __DSVERIFIER_assert(check_settling_time(A, B, C, D, X0, u, tsr, p, ts, nStates) == 0);
+//  __ESBMC_assert(check_settling_time(A, B, C, D, X0, u, tsr, p, ts, _controller.nStates) == 0);
```

Comment:
Do we really need these commented lines?

Remember that we've version control.